rosbag2_cpp/include/rosbag2_cpp/readers/sequential_reader.hpp

```diff
@@ -68,6 +69,8 @@ class ROSBAG2_CPP_PUBLIC SequentialReader
 
   std::vector<rosbag2_storage::TopicMetadata> get_all_topics_and_types() override;
 
+  void set_filter(const rosbag2_storage::StorageFilter & storage_filter);
```

Comment:
Do you think it makes sense to introduce a `reset_filter` function? Or generally the use case of I first want to read all topics `a, b, c` then want to apply a different filter for `d, e, f`.
I guess you could achieve this by simply overriding the filter, but I wonder if there are use cases where you'd want to reset the applied filter.

Reply:
Added in 7d16787